idaes/models_extra/power_generation/unit_models/soc_submodels.py

```diff
@@ -0,0 +1,3997 @@
+        
+        if not self.config.below_electrode:
+            self.Dconc_x0 = pyo.Var(tset, iznodes, comps,
+                doc="Deviation of concentration at electrode surface from that "
+                "in the bulk channel",
+                initialize=0,
+                units = pyo.units.mol/pyo.units.m**3
+                )
+            self.xflux_x0 = pyo.Var(tset, iznodes, comps,
+                doc="Material flux from electrode surface to channel "
+                "(positive is in)",
+                initialize=0,
+                units = pyo.units.mol/pyo.units.m**2/pyo.units.s
+                )
+            @self.Expression(tset,iznodes,comps)
+            def Dconc_x1(b,t,iz,j):
+                return 0
+            @self.Expression(tset,iznodes,comps)
+            def xflux_x1(b,t,iz,j):
+                return 0
+        else:
+            self.Dconc_x1 = pyo.Var(tset, iznodes, comps,
+                doc="Deviation of concentration at electrode surface from that "
+                "in the bulk channel",
+                initialize=0,
+                units = pyo.units.mol/pyo.units.m**3
+                )
+            self.xflux_x1 = pyo.Var(tset, iznodes, comps,
+                doc="Material flux from channel to electrode surface "
+                "(positive is out)",
+                initialize=0,
+                units = pyo.units.mol/pyo.units.m**2/pyo.units.s
+                )
+            @self.Expression(tset,iznodes,comps)
+            def Dconc_x0(b,t,iz,j):
+                return 0
+            @self.Expression(tset,iznodes,comps)
+            def xflux_x0(b,t,iz,j):
+                return 0
+        # Channel thickness AKA length in the x direction is specific to the
+        # channel so local variable here is the only option
+        self.length_x = pyo.Var(
+            doc="Thickness from interconnect to electrode (x-direction)",
+            units=pyo.units.m,
+        )
+        self.heat_transfer_coefficient = pyo.Var(
+            tset,
+            iznodes,
+            doc="Local channel heat transfer coefficient",
+            initialize=500,
+            units=pyo.units.J / pyo.units.m ** 2 / pyo.units.s / pyo.units.K,
+        )
+        self.flow_mol = pyo.Var(
+            tset,
+            iznodes,
+            doc="Molar flow in the z-direction through faces",
+            units=pyo.units.mol / pyo.units.s,
+            bounds = (-1E-9, None)
+        )
+        self.conc = pyo.Var(
+            tset,
+            iznodes,
+            comps,
+            doc="Component concentration at node centers",
+            units=pyo.units.mol / pyo.units.m ** 3,
+            bounds = (0,None)
+        )
+        self.Dtemp = pyo.Var(
+            tset,
+            iznodes,
+            doc="Deviation of temperature at node centers "
+                "from temperature_z",
+            units=pyo.units.K,
+            bounds = (-1000,1000)
+        )
+        self.enth_mol = pyo.Var(
```

Comment:
A lot of this looks like it is doing things in the unit model that would normally be done in a `StateBlock`. Whilst it is more work to set up, I would suggest considering creating a (very simple) property package that will support these and using the more standard IDAES approach in order to facilitate future maintenance of the code and eventual promotion to the core model libraries.

Reply:
While I don't disagree that may need to be saved for the future.  We can probably swap that put pretty seamlessly by making a new default prop pack.